lib/paypal_express/api.rb

```diff
@@ -84,37 +85,32 @@ def refund_payment(kb_account_id, kb_payment_id, kb_payment_transaction_id, kb_p
       end
 
       def get_payment_info(kb_account_id, kb_payment_id, properties, context)
-        ignored_api_calls = [:details_for]
-        responses = @response_model.from_kb_payment_id(@transaction_model, kb_payment_id, context.tenant_id)
-        responses = responses.reject do |response|
-          ignored_api_calls.include?(response.api_call.to_sym)
-        end
-        t_info_plugins = responses.collect do |response|
-          response.to_transaction_info_plugin(response.send("#{@identifier}_transaction"))
-        end
+        filtered_plugin_info, plugin_info, with_only_pending_trx = get_raw_payment_info(kb_payment_id, context)
         # Should never happen...
-        return [] if t_info_plugins.nil?
+        return [] if filtered_plugin_info.nil?
```

Comment:
Good catch. That case should never happen though. Modified the codes and added a test for that. See 5da55f6